src/test/java/uk/gov/pay/products/resources/ProductResourceIT.java

```diff
@@ -918,4 +921,50 @@ public void shouldReturn400_whenSavingProduct_withReferenceEnabledAndNoReference
                 .then()
                 .statusCode(400);
     }
+
+    @Test
+    public void shouldReturnMetadata_whenMetadataExistsForAProduct() {
+        String externalId = randomUuid();
+        int gatewayAccountId = randomInt();
+
+        Product product = ProductEntityFixture.aProductEntity()
+                .withExternalId(externalId)
+                .withGatewayAccountId(gatewayAccountId)
+                .build()
+                .toProduct();
+
+        databaseHelper.addProduct(product);
+
+        databaseHelper.addMetadata(externalId, "key", "value");
+
+        ValidatableResponse response = givenSetup()
+                .when()
+                .accept(APPLICATION_JSON)
+                .get(format("/v1/api/products/%s", externalId))
+                .then()
+                .statusCode(200);
+
+        Integer intPrice = response.extract().path(PRICE);
+        Long price = Long.valueOf(intPrice);
+        assertThat(price, equalTo(product.getPrice()));
+
+        response
+                .body(NAME, is(product.getName()))
+                .body(GATEWAY_ACCOUNT_ID, is(gatewayAccountId))
+                .body(TYPE, is(product.getType().name()))
+                .body(DESCRIPTION, is(product.getDescription()))
+                .body(RETURN_URL, is(product.getReturnUrl()))
+                .body(METADATA + "[0].key",is("value"));
+
+        String productsUrl = "https://products.url/v1/api/products/";
+        String productsUIPayUrl = "https://products-ui.url/pay/";
+        response
+                .body("_links", hasSize(2))
+                .body("_links[0].href", matchesPattern(productsUrl + externalId))
+                .body("_links[0].method", is(HttpMethod.GET))
+                .body("_links[0].rel", is("self"))
+                .body("_links[1].href", matchesPattern(productsUIPayUrl + externalId))
+                .body("_links[1].method", is(HttpMethod.GET))
+                .body("_links[1].rel", is("pay"));
```

Comment:
Assertion for metadata ? as test specifically says `shouldReturnMetadata`

Reply:
exists on line 957, can remove this at is comes from another test

Reply:
`                .body(METADATA + "[0].key",is("value"));`